include/aws/common/json.h

```diff
@@ -0,0 +1,358 @@
+ * @return AWS_OP_SUCCESS if the aws_json_value is a number aws_json_value, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+bool aws_json_is_number(const struct aws_json_value *node);
+
+/**
+ * Checks if the aws_json_value is a array.
+ * @param node The aws_json_value to check.
+ * @return AWS_OP_SUCCESS if the aws_json_value is a array aws_json_value, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+bool aws_json_is_array(const struct aws_json_value *node);
+
+/**
+ * Checks if the aws_json_value is a boolean.
+ * @param node The aws_json_value to check.
+ * @return AWS_OP_SUCCESS if the aws_json_value is a boolean aws_json_value, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+bool aws_json_is_boolean(const struct aws_json_value *node);
+
+/**
+ * Checks if the aws_json_value is a null aws_json_value.
+ * @param node The aws_json_value to check.
+ * @return AWS_OP_SUCCESS if the aws_json_value is a null aws_json_value, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+bool aws_json_is_null(const struct aws_json_value *node);
+
+/**
+ * Checks if the aws_json_value is a object aws_json_value.
+ * @param node The aws_json_value to check.
+ * @return AWS_OP_SUCCESS if the aws_json_value is a object aws_json_value, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+bool aws_json_is_object(const struct aws_json_value *node);
+// ====================
+
+// ====================
+// Memory Management
+
+/**
+ * Initializes the JSON module for use.
+ * @param allocator The allocator to use for creating aws_json_value structs.
+ */
+AWS_COMMON_API
+void aws_json_module_init(const struct aws_allocator *allocator);
+
+/**
+ * Cleans up the JSON module. Should be called when finished using the module.
+ */
+AWS_COMMON_API
+void aws_json_module_cleanup(void);
+
+/**
+ * Removes the aws_json_value from memory. If the aws_json_value is a object or array, it will also destroy
+ * attached aws_json_values as well.
+ *
+ * For example, if you called "aws_json_array_add(b, a)" to add an object "a" to an array "b", if you call
+ * "aws_json_destroy(b)" then it will also free "a" automatically. All children/attached aws_json_values are freed
+ * when the parent/root aws_json_value is destroyed.
+ * @param node The aws_json_value to destroy.
+ * @return AWS_OP_SUCCESS if the destroy was successful, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+int aws_json_destroy(const struct aws_json_value *node);
+// ====================
+
+// ====================
+// Utility
+
+/**
+ * Returns an unformatted JSON string representation of the aws_json_value.
+ *
+ * Note: When you are finished with the aws_byte_buf, you must call "aws_byte_buf_clean_up_secure" to free
+ * the memory used, as it will NOT be called automatically.
+ * @param node The aws_json_value to format.
+ * @param output The destination for the JSON string
+ * @return A string containing the JSON.
+ */
+AWS_COMMON_API
+int aws_json_to_string(const struct aws_json_value *node, struct aws_byte_buf *output,
```

Comment:
These shouldn't take an allocator.  Use the byte_buf APIs to either write or append the value into the buffer.